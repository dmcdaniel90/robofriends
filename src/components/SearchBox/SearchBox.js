import React from "react";

const SearchBox = ({ searchChange }) => {
  return (
    <div className="pa2" data-testid="search-box">
      <label htmlFor="search" hidden>Search Robots</label>
      <input
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        placeholder="Search Robots"
        onChange={searchChange}
        id="search"
      />
    </div>
  )
}

export default SearchBox;