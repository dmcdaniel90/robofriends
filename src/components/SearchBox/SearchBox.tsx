import {ChangeEventHandler} from 'react';

type SearchBoxProps = {
  searchChange: ChangeEventHandler<HTMLInputElement>
}

const SearchBox = ({ searchChange }: SearchBoxProps) => {
  return (
    <div className="pa2" data-testid="search-box">
      <label htmlFor="search" hidden>Search Robots</label>
      <input
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        placeholder="Search Robots"
        onChange={searchChange}
        id="search"
      />
    </div>
  )
}

export default SearchBox;