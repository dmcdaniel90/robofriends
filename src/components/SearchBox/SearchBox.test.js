import {render} from '@testing-library/react'
import SearchBox from './SearchBox';

describe('SearchBox', () => {
  it('expect to render SearchBox component', () => {
    const {container} = render(<SearchBox />)
    expect(container).toMatchSnapshot();
  })
})