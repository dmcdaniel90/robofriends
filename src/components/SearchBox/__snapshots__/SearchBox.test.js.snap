// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox expect to render SearchBox component 1`] = `
<div>
  <div
    class="pa2"
    data-testid="search-box"
  >
    <label
      for="search"
      hidden=""
    >
      Search Robots
    </label>
    <input
      class="pa3 ba b--green bg-lightest-blue"
      id="search"
      placeholder="Search Robots"
      type="search"
    />
  </div>
</div>
`;
